### PR TITLE
Y24 141 labwhere reception to scan tubes and plates into locations

### PR DESCRIPTION
**Pull request by seenanair**
Closes https://github.com/sanger/traction-ui/issues/1773

#### Changes proposed in this pull request

- Adds Labwhere reception page to store labwhere barcodes 
   - This includes a form interface for users to scan in/out labware barcodes to/from a specified location in Labwhere.
   -  The form includes fields for: User barcode or swipecard, Location barcode, Start position (optional), Labware barcodes
   - The component validates the input fields for 'User barcode', and 'Labware barcodes'
       and displays error messages for any invalid or missing data when the form is submitted or when the input fields are 
        updated.
   - The 'Start position' field is optional and does not require validation.
   -  Upon successful validation, the form data is submitted to the Labwhere API to store the barcodes in the specified 
         location.
   - It displays a success message if the barcodes are stored successfully, or an error message if the submission fails.

- Adds `scanBarcodesIntoLabwhereLocation` helper method for storing barcodes to Labwhere location using Labwhere `scans` api
- Updates `FetchWrapper` class to have content_type parameter

#### Instructions for Reviewers
Note : @BenTopping @stevieing 
During testing, it was realised that there is currently no validation for barcodes in LabWhere, allowing any text entered as a labware barcode to be stored in a location.  This raises the question of whether we need separate validation in Traction for invalid barcodes using the Sequencescape API, which requires further discussion and review.
(In similar cases, such as when using `scans` api in AssetAudits, barcode validation is handled through the Sequencescape API.)

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: tests/unit/views/LabWhereReception.spec.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    const barcodeInput = wrapper.find('#labware_barcodes')
+    await barcodeInput.setValue('barcode1')
+
+    await wrapper.vm.$nextTick()
+    expect(wrapper.vm.errors).toEqual({
+      user_code: 'User code is required',
+    })
+
+    const [userInput] = wrapper.findAll('input')
```
Comment (BenTopping) [inline tests/unit/views/LabWhereReception.spec.js:75]: Worth using a test-id or data attribute to make input more discoverable?

File: src/router.js [hunk 2]
```diff
@@ -53,6 +54,12 @@ const router = createRouter({
       meta: { pipeline: 'Reception' },
       component: GeneralReception,
     },
+    {
+      path: '/labwhere-reception',
+      name: 'LabwhereReception',
+      meta: { page: 'Labwhere Reception - Scan In/Out' },
```
Comment (BenTopping) [inline src/router.js:60]: LabWhere*

**Review by BenTopping — CHANGES_REQUESTED** [review]
Good start, tested locally and the page looks good and works well with nice error handling 👌 

Couple things:
- application/json should work as the content-type for the scans api. Ive tested locally with a basic CURL and seems to work so might be a setup issue for the CORS error.
- Im not sure the preview area of the page adds much. Maybe a text highlight in red for duplicate barcodes and an error preventing submission if there are duplicates would be better.
- A clear button to remove the labware location, position and barcodes might be useful if they are doing a lot of labware at once (not sure if this will happen but worth thinking about).

**Comment by seenanair** [comment]
> Good start, tested locally and the page looks good and works well with nice error handling 👌
> 
> Couple things:
> 
> * application/json should work as the content-type for the scans api. Ive tested locally with a basic CURL and seems to work so might be a setup issue for the CORS error.

Oh, interesting. I tried it many different ways, but never worked for me with CORS error.

> * Im not sure the preview area of the page adds much. Maybe a text highlight in red for duplicate barcodes and an error preventing submission if there are duplicates would be better.

Following was my thought to do a preview area. 'Labware count' is specified as a requirement by the stakeholder, and I believe it would be useful to display it as preview text. Additionally, since we are not validating against an empty location barcode, which could potentially cause a reverse effect (scanning out) on normal operations, it is important to display the intended action's purpose somewhere for confirmation.

> * A clear button to remove the labware location, position and barcodes might be useful if they are doing a lot of labware at once (not sure if this will happen but worth thinking about).
A 'Reset' button was initially there but in attempt to make it similar to LabWhere scan in page, I removed it.

**Comment by seenanair** [comment]
> > Good start, tested locally and the page looks good and works well with nice error handling 👌
> > Couple things:
> > 
> > * application/json should work as the content-type for the scans api. Ive tested locally with a basic CURL and seems to work so might be a setup issue for the CORS error.
> 
> @BenTopping Oh, interesting. I tried it many different ways, but never worked for me with CORS error.
> 
> > * Im not sure the preview area of the page adds much. Maybe a text highlight in red for duplicate barcodes and an error preventing submission if there are duplicates would be better.
> 
> Following was my thought to do a preview area. 'Labware count' is specified as a requirement by the stakeholder, and I believe it would be useful to display it as preview text. Additionally, since we are not validating against an empty location barcode, which could potentially cause a reverse effect (scanning out) on normal operation of scanning in, thought it is important to display the intended action's purpose somewhere for confirmation.
> 
> > * A clear button to remove the labware location, position and barcodes might be useful if they are doing a lot of labware at once (not sure if this will happen but worth thinking about).
> 
> A 'Reset' button was initially there but in attempt to make it similar to LabWhere scan in page, I removed it. :)

**Review by BenTopping — APPROVED** [review]

**Pull request merged**